R/daisyTimeline.R

```diff
@@ -2,14 +2,44 @@
 #'
 #' Create a Daisy Timeline
 #'
+#' @param events A data frame with columns 'date' and 'content', and optionally 'side'
+#' @param width Width of the widget
+#' @param height Height of the widget
+#' @param elementId HTML element ID
+#'
 #' @import htmlwidgets
 #'
 #' @export
 daisyTimeline <- function(events, width = NULL, height = NULL, elementId = NULL) {
-  # events is a list of lists: each item has date, content, maybe side
+  # Convert data frame to list of lists for JavaScript consumption
+  if (is.data.frame(events)) {
+    # Check required columns
+    if (!all(c("date", "content") %in% names(events))) {
+      stop("Data frame must contain 'date' and 'content' columns")
+    }
+    
+    # Convert data frame to list of lists
+    events_list <- lapply(seq_len(nrow(events)), function(i) {
+      event <- list(
+        date = as.character(events$date[i]),
+        content = as.character(events$content[i])
+      )
+      
+      # Add side column if it exists
+      if ("side" %in% names(events)) {
+        event$side <- as.character(events$side[i])
+      }
+      
+      return(event)
+    })
```

Comment:
This also seems convoluted - we can surely do this without cycling through each row?